### DISCUSSION TITLE
Masks for the images

### DISCUSSION THREAD

**Discussion by Anudeepsrib** [alzayats/DeepFish#9] [Q&A] [answered]
Hi, if I have to try and validate the model on an external dataset, I understand that I will have to input masks. Is there any code or technique to create the masks like the ones you had for your input images? Please throw some light on this. Your work is quite interesting 👍

**Comment by alzayats** [comment]
Hi, 
Our model's input format is binary masks.
Is your external dataset already has masks and you want to convert it to binary masks or not?
For example:
To convert COCO masks to binary masks. https://stackoverflow.com/questions/50805634/how-to-create-mask-images-from-coco-dataset
To create your own. https://labelstud.io/

**Reply by Anudeepsrib** [reply]
No, the external dataset that I was trying to use doesn't have any masks. I tried photoshop to make binary masks for each image, but it is definitely not an ideal way to deal with 1 GB of the fish dataset. 

I tried googling and using different codes and tools to get binary masks, but the binary mask I made has 40 points on the ground truth, while there is only one fish in the image.

Below is a sample prediction for an external image using predict.py where I used a binary mask that I generated using photoshop and Matlab.

Please validate my below understanding and kindly correct me if I am wrong:
- I am assuming the number of points in the mask should be 1 when there are only 1 fish in the input image

- Based on your previous reply to another developer, I am assuming that the predict.py would use a pre-trained model (model trained on your dataset) to predict the number of fish in the external image and its location. Kindly correct me in this regard too. 

![11](https://user-images.githubusercontent.com/36981925/208266488-fe8594a5-d0c3-4930-8b49-a64e4286024b.png)


-------
Grateful for your time 👍

**Comment by alzayats** [comment, marked as answer]
Try this tool to get binary masks  https://lear.inrialpes.fr/people/klaeser/software_image_annotation
or this https://labelstud.io/

your questions are correct
points in the mask should be 1 when there are only 1 fish in the input image
the predict.py would use a pre-trained model

**Reply by Anudeepsrib** [reply]
Thank you so much for the confirmation and help with the masks 👍
